scipy/optimize/tests/test_zeros.py

```diff
@@ -1246,3 +1247,342 @@ def test_maxiter_int_check_gh10236(method):
     message = "'float' object cannot be interpreted as an integer"
     with pytest.raises(TypeError, match=message):
         method(f1, 0.0, 1.0, maxiter=72.45)
+
+
+class TestDifferentiate():
```

Comment:
We should add some tests for evaluating the derivative of a function `f` at a saddle point. I like the examples

```python
def f(x):
    return (x - 1)**3

def g(x):
    return np.where(x > 1, (x - 1)**5, (x - 1)**3)
```

evaluated at `x = 1`.

Reply:
Done - with an absolute tolerance for now.